sw/host/provisioning/ft/src/main.rs

```diff
@@ -119,19 +112,44 @@ fn main() -> Result<()> {
     let spi_console_device = SpiConsoleDevice::new(&*spi)?;
     InitializeTest::print_result("load_bitstream", opts.init.load_bitstream.init(&transport))?;
 
-    // Format test tokens.
+    // Parse and format LC tokens.
     let _test_unlock_token =
         hex_string_to_u32_arrayvec::<4>(opts.provisioning_data.test_unlock_token.as_str())?;
     let _test_exit_token =
         hex_string_to_u32_arrayvec::<4>(opts.provisioning_data.test_exit_token.as_str())?;
     let rma_unlock_token =
         hex_string_to_u32_arrayvec::<4>(opts.provisioning_data.rma_unlock_token.as_str())?;
-    // Format ujson data payload(s).
-    // Individualization ujson payload.
+
+    // Parse and prepare individualization ujson data payload.
     let _ft_individualize_data_in = ManufFtIndividualizeData {
         device_id: hex_string_to_u32_arrayvec::<8>(opts.provisioning_data.device_id.as_str())?,
     };
-    // Personalization ujson payload.
+
+    // Parse and prepare CA key.
+    let mut ca_cfgs: HashMap<String, CaConfig> = from_str(
```

Comment:
Since `from_str` is a bare function in the `serde_json` namespace, it'd be better to have the full "path" to the function here:
```
let foo = serde_json::from_str( ... )?;
```

Also, the docstring for `ca_config` says "CA HJSON config file", but by using `serde_json`, this is strict json.

Should do one of:
- Support hjson via `deser_hjson::from_str`
- Support json, json5 or hjson via `serde_annotate::from_str`
- Support only strict json and change the docstring.

Reply:
Fixed. Went with serde_annotate for most flexibility in the future.